### ISSUE TITLE
Deprecated EntityCreatePortalEvent

### ISSUE THREAD

**Issue by SlimeDog** [Multiverse/Multiverse-Core#2000]
Spigot 1.14.2 (Jenkins 2323) git-Spigot-03bd4b0-342194e (MC: 1.14.1) (Implementing API version 1.14.1-R0.1-SNAPSHOT)
WorldGuard 7.0.0-SNAPSHOT;1889-2b854f0
Multiverse-Core 3.0.0-SNAPSHOT-b761

On server start-up:
```
[07:34:59] [Server thread/WARN]: [Multiverse-Core] "Multiverse-Core v3.0.0-SNAPSHOT-b761" has registered a listener for org.bukkit.event.entity.EntityCreatePortalEvent on method "public void com.onarandombox.MultiverseCore.listeners.MVPortalListener.entityPortalCreate(org.bukkit.event.entity.EntityCreatePortalEvent)", but the event is Deprecated. "Server performance will be affected"; please notify the authors [Rigby, fernferret, lithium3141, main--, dumptruckman].
```

**Comment by MonstahGames** [comment]
Yep, having this too.

**Comment by 0xC1A551C** [comment]
same

**Comment by SlimeDog** [comment]
It's universal. No reason to keep adding "me too". :)

**Comment by GCD56** [comment]
Tried the latest available build: here is the full text from the Paper Server startup...
(Posted as reference to others looking to post this issue, not to bug the developers)
Multiverse-Core v4.0.0-SNAPSHOT-b764" has registered a listener for org.bukkit.event.entity.EntityCreatePortalEvent on method "public void com.onarandombox.MultiverseCore.listeners.MVPortalListener.entityPortalCreate(org.bukkit.event.entity.EntityCreatePortalEvent)", but the event is Deprecated. "Server performance will be affected"; please notify the authors [Rigby, fernferret, lithium3141, main--, dumptruckman]

**Comment by Jadiction** [comment]
I noticed I forgot to inform anyone viewing this that Multiverse is aware of the deprecated event that you're informed about in console. The "performance impact" isn't noticeable and is fine to ignore.

**Comment by Andre601** [comment]
Since this seems to be a known thing I guess I don't have to mention that I have this too?

**Comment by SlimeDog** [comment]
Nope. You didn't have to mention it. :)

**Comment by GCD56** [comment]
And yet this issue is closed while not being addressed at all. Has to make one wonder what else might be in the project that isn't so obvious. This is very shoddy methodology. Makes one wonder if these guys are part of the "elite" that loves to tell beginners they know the answer but refuse to spoon feed. I've always felt the real reason for this attitude and a lack of a fix in cases like this are both due to having no clue what to do and an inability to admit it? No decent programmer would ignore their users and leave known errors that reflect badly or the author's abilities if they could actually do something about the problem. Lame and pitiful.

**Comment by dumptruckman** [comment]
@GCD56 First of all, this issue is not closed. Secondly, we've acknowledged this "issue" on many occasions. Look at all the duplicate tickets we have closed linking to this one. Jadiction has even responded to this ticket explaining that we're aware of this "issue" and that it is something you can completely ignore as the performance impact is negligible.

The "issue" has not been fixed yet because the Spigot team removed a *vital* API from Bukkit in 1.14 yet they have stated they intend to add it back. Because of the removal of this API, We've had to build MV against 1.13 and create a hack to make the nether portal functionality work in 1.14. If we build against 1.14, we will lose support for 1.13. The only way to remove this warning is to build against 1.14 and use a new event to replace the deprecated event. Unfortunately, this new event does not exist in 1.13.

So, we're aware of the issue, we're aware of how to fix it, and we're also aware that there's no harm in not fixing it (besides having to deal with a ridiculous amount of people complaining about it.) Therefore, we are waiting for the Bukkit API situation to improve (and also the 1.14 performance issues to be fixed) so we can feel comfortable dropping support for 1.13.

p.s.

I am the sole developer on this project and I work on it and provide support for it for free*. I do not play MC anymore so I think you should be happy MV is even still being maintained in the slightest. I am a father, I work a full time job, and I like to have time doing things I enjoy when I can. I often do not have time to look at every single issue ticket that comes in (and there's a **lot** of them). However, I am often very responsive on Discord and I encourage anyone to come hang out in the MV discord and chat.

\* I have recently started a [Patreon page](https://www.patreon.com/dumptruckman) so people can help support my efforts on this project. However, even then I'm planning to use those funds to afford stuff for the MV project.

**Comment by GCD56** [comment]
Now THAT was a reasonable response. First one I’ve seen that explained the issue. (Not doubting there have been others, just saying this is the first I’ve seen.)
Now the facts are perfectly clear for all you have my sincere apology as well. I'm too used to being blown off and watching others be blown off with snobbery, smart remarks, and rude answers. This thread was all I had read and it felt like more of the same. Being a dad myself I get the lack of time. Again, My apologies.

**Comment by Jayderp** [comment]
I don't think this needs to be fixed, however, it might scare some people.

**Comment by SlimeDog** [comment]
> ...2 identical servers...
Same Java version?

**Comment by GCD56** [comment]
I did say warning but as far as I'm concerned if it shouldn't happen its an error. The only difference is severity. Nit picked? And yes, identical sort of means "the same".

**Comment by SlimeDog** [comment]
Please back off. "Identical servers" is ambiguous. It could refer to the Spigot/MC servers, the underlying environment (OS and Java and etc), hardware, some of that list, or all of that list. It is reasonable to ask for clarification. Thanks for confirming that you meant "the entire system".

**Comment by GCD56** [comment]
In every way meaningful to this topic they are identical.
For the lawyers out there is means:
Same Physical server box
Same underlying game configuration
Same software version
Copies of the same maps
All the same plugins and all on their same versions.
Same keyboard
Same mouse
Same operator
Same room
Same town
Started at as close to the same time as reasonably possible
Same Same Same

OR we could just say "Identical Servers", so much simpler.

**Comment by SlimeDog** [comment]
Yes, we got it. Keep your lawyer pants on. You're perfect, just suffering an anomaly.

I cannot begin to count the number of times in support situations that "identical" has meant something like "completely the same ... except (oops!) I forgot to mention that one is running 1.14.1 and the other 1.14.4. But **otherwise** the same." So the confirmation was useful. Thanks again.

**Comment by GCD56** [comment]
All, 
I decided to get to the bottom of why I saw this message on 1 of 2 "identical" servers.
I compared line by line all of multiverse config and world files. all matched exactly.
Made no sense. So I backed up and began looking at other configuration files and I found it.

In Bukkit.yml look for this line:
deprecated-verbose: default
change to 
deprecated-verbose: false
Doesn't "fix" the depreciated code but will at least remove the message.

**Comment by SlimeDog** [comment]
Thanks for reporting.

**Comment by dumptruckman** [comment]
@GCD56 nice find!

**Comment by andrewkm** [comment]
Same thing here.

**Comment by GCD56** [comment]
@EbonJaeger
Did you not read the thread before posting?
They can't fix this right now, apparently the event was depreciated without an alternative.
It wouldn't be the first time. So I posted the work around if the message bugs you:

In Bukkit.yml look for this line:
deprecated-verbose: default
change to
deprecated-verbose: false
Doesn't "fix" the depreciated code but will at least remove the message.

**Comment by EbonJaeger** [comment]
@GCD56 What in the world are you on about? Was simply linking the duplicate issue to this one.

**Comment by x9nico** [comment]
Hi, it does the same thing in PaperSpigot 1.15.

**Comment by Nogapra** [comment]
Hi, it does the same thing in PaperSpigot (build 26) 1.15.1

**Comment by GCD56** [comment]
So set the flag to false and you won’t have to see it.

**Comment by misterwidescreen** [comment]
WARN "Multiverse-Core v4.1.0-b775" has registered a listener for org.bukkit.event.entity.EntityCreatePortalEvent on method "public void com.onarandombox.MultiverseCore.listeners.MVPortalListener.entityPortalCreate(org.bukkit.event.entity.EntityCreatePortalEvent)", but the event is Deprecated. "Server performance will be affected"; please notify the authors [Rigby, fernferret, lithium3141, main--, dumptruckman].

**Comment by benwoo1110** [comment]
`[11:10:07] [Server thread/INFO]: [Multiverse-Core] We are aware of the warning about the deprecated event. There is no alternative that allows us to do what we need to do. The performance impact is negligible.
`
The line right after that warning ^^

This is not an issue, and you do not worry about it. Thanks!

**Comment by misterwidescreen** [comment]
@benwoo1110 You are totally right! I only saw the warning and thought "I had to report" haha xD guys you already know this just a reminder then a guess Kappa. Stay safe and healthy!

**Comment by ljacqu** [comment]
> just a reminder then

Yep makes absolutely sense to post the error without any comment in an issue that explains in depth why it's fine

**Comment by Meledosia** [comment]
You know the programmer doesn’t even play Minecraft any more right?

He maintains Multi-Verse as a kindness, do you really want to tick him off?

You think you have the right to possibly screw us all over something that isn’t even impacting your servers???

 

The error says all it can in the space available: A depreciated call that has yet to be replaced with something better.

 

Meledosia

**Comment by misterwidescreen** [comment]
@ljacqu and @Meledosia I understood after benwoo commented. I am new to this github and server stuff. The only thing I am reading when there are problems are to post the error log. Thats why I posted the error log but I saw after I posted my "error" when I red the description and saw the info I missed at first that this was not something that would impact.  No need for you guys to post and act like this I got the point for 3 days ago and your comments right now are also valued at 0 mine had atleast the error log. I thought I did a good thing to report it and not get flamed. But I red get the lession read even more and understand it before posting. If I just would have seen the INFO in log I wouldn't even find this. 

So the authors' if you see this I am sry that i posted an error that isn't an error , and sry if I took some time of your life if you would have been reading this. Keep up the awesome work I am very greatfull for your effort to the community and you are doing an greatjob. Really great plugin! Stay safe in this pandemi and take care of eachother.

Best Regards

**Comment by Gryffin78** [comment]
[Server thread/INFO]: This server is running CraftBukkit version git-Spigot-758abbe-3f0c333 (MC: 1.16.1) (Implementing API version 1.16.1-R0.1-SNAPSHOT)
[Server thread/INFO]: [WorldGuard] Loading WorldGuard v7.0.4-beta1;bc551da
[Server thread/INFO]: [Multiverse-Core] Enabling Multiverse-Core v4.1.0-b775
[Server thread/WARN]: [Multiverse-Core] Loaded class net.milkbowl.vault.economy.Economy from Vault v1.7.2-b107 which is not a depend, softdepend or loadbefore of this plugin.
[Server thread/WARN]: [Multiverse-Core] "Multiverse-Core v4.1.0-b775" has registered a listener for org.bukkit.event.entity.EntityCreatePortalEvent on method "public void com.onarandombox.MultiverseCore.listeners.MVPortalListener.entityPortalCreate(org.bukkit.event.entity.EntityCreatePortalEvent)", but the event is Deprecated. "Server performance will be affected"; please notify the authors [Rigby, fernferret, lithium3141, main--, dumptruckman].
[Server thread/INFO]: [Multiverse-Core] We are aware of the warning about the deprecated event. There is no alternative that allows us to do what we need to do. The performance impact is negligible.

**Comment by benwoo1110** [comment]
For all who see this issue: 
**[Server thread/INFO]: [Multiverse-Core] We are aware of the warning about the deprecated event. There is no alternative that allows us to do what we need to do. The performance impact is negligible.**

Thank you for your understanding.

**Comment by TheDGOfficial** [comment]
The server performance will be affected thing mainly exists because when AsyncPlayerChatEvent added to Bukkit, PlayerChatEvent was using some old tech causing messages to be delayed or such. This event does not affect performance, but a better alternative exists, so it is deprecated.

Even if it doesn't affect anything, deprecated things will eventually removed, and they should be switched to better alternatives. 

Looking at the source, this can just be replacted with PortalCreateEvent on new versions and checking the getEntity method does not return null. This way it can ensure it has same behaviour as EntityPortalCreateEvent. (This should only be done on new versions though, older ones still have PortalCreateEvent but they do not have a getEntity method)

**Comment by nicegamer7** [comment]
PlayerChatEvent does not have anything to do with the warning the server gives on startup. Regardless, we are aware of both deprecated events being used and will be removing them in MV 5.

**Comment by TheDGOfficial** [comment]
> PlayerChatEvent does not have anything to do with the warning the server gives on startup. Regardless, we are aware of both deprecated events being used and will be removing them in MV 5.

I said that the "performance will be affected" thing was added to deprecated message when PlayerChatEvent deprecated on Bukkit. (historical thing)

**Comment by benwoo1110** [comment]
This is fixed in mv5
